File: Unit1/c-programs/Programa181/v2/consumidor.c
#include <stdio.h>
#include "fichero.h"

int main(int argc,char **argv){
  int contador;
  contador=leovalor();
  printf(" Consumidor: Lectura=%d \n",contador);
  contador--;
  escribovalor(contador);
  printf(" Consumidor: Escritura=%d \n",contador);
}

File: Unit1/java-programs/Programa 1.8.3/src/Consumidor.java

public class Consumidor extends Thread{
  public Consumidor(String Nombre) {
    if (Nombre!=null) setName(Nombre);
	start();
  }
  public void run(){
    int contador;
	contador=Fichero.leovalor();
	System.out.printf(" Consumidor: Lectura=%d \n",contador);
	contador--;
	Fichero.escribovalor(contador);
	System.out.printf(" Consumidor: Escritura=%d \n",contador);
  }
}


